turbinia/evidence.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Turbinia Evidence objects."""
+
+import json
+
+
+class Evidence(object):
+  """Evidence object for processing.
+
+  In most cases, these objects will just contain metadata about the actual
+  evidence.
+  """
+
+  def __init__(self, name=None, description=None, source=None, local_path=None,
+               tags=None):
+    """Initialization for Evidence.
+
+    Args:
+      name: Name of evidence.
+      description: Description of evidence.
+      source: String indicating where evidence came from (including tool version
+              that created it, if appropriate).
+      local_path: A string of the local_path to the evidence.
+      tags: dict of extra tags assocated with this evidence.
+    """
+    self.name = name
+    self.local_path = local_path
+    self.source = source
+    self.description = description
+    self.tags = tags if tags else {}
+
+    # List of jobs that have processed this evidence
+    self.processed_by = []
```

Comment:
Good question.  The evidence does get passed back to the jobs, but as of right now only the newly created Evidence is passed back.  I'll have to figure out how to plumb this in, otherwise I might actually remove it.